src/CycloneDX.Core/Models/Hash.cs

```diff
@@ -62,5 +62,40 @@ public enum HashAlgorithm
         [XmlText]
         [ProtoMember(2)]
         public string Content { get; set; }
+
+        public bool Equivalent(Hash obj)
+        {
+            return (!(obj is null) && this.Alg == obj.Alg);
+        }
+
+        public bool MergeWith(Hash obj)
+        {
+            try
+            {
+                // Basic checks for null, type compatibility,
+                // equality and non-equivalence; throws for
+                // the hard stuff to implement in the catch:
+                return base.MergeWith(obj);
+            }
+            catch (BomEntityConflictException)
+            {
+                // Note: Alg is non-nullable so no check for that
+                if (this.Content is null && !(obj.Content is null))
+                {
+                    this.Content = obj.Content;
```

Comment:
Should we really just inject the content of the other hash (and thus just assume that the hashes would be the same)?

Reply:
Well, got a couple of answers to this situation:

1) As commented in the PR, the hassle with hashes is more of an illustration for the new mechanism which fits in 1-2 screenfulls - unlike the larger change for Components (even if someone finds a way to externalize much of that code, there are still quite a few use-cases to handle there).

2) This particular clause here takes a `Hash this` instance that has an `Alg` (non-nullable enum) value, but no `Content` value. I am not sure TBH if those can appear in reality outside deliberately screwed test cases. If the other object we are merging into `this` does have a non-null `Content` (hash string) for the algo, we accept that. Otherwise falling through to the next check - if the two contents for same algo mismatch, throw an exception. We know the algos matched because two `Hash` objects are `Equivalent()` by our definition, and checked in the `base.MergeWith()` which would `return false` for unrelated objects.